plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java

```diff
@@ -851,6 +854,106 @@ public void testTableWithNonNullableColumns()
+    {
+        String tableName = "test_create_table_with_reader_writer_version_" + randomNameSuffix();
+
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (reader_version = 0)", SCHEMA, tableName),
+                "reader_version must be an integer between \\[1, 2\\]");
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (reader_version = 3)", SCHEMA, tableName),
+                "reader_version must be an integer between \\[1, 2\\]");
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (writer_version = 1)", SCHEMA, tableName),
+                "writer_version must be an integer between \\[2, 4\\]");
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (writer_version = 5)", SCHEMA, tableName),
+                "writer_version must be an integer between \\[2, 4\\]");
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (change_data_feed_enabled = true, writer_version = 1)", SCHEMA, tableName),
+                "writer_version must be an integer between \\[2, 4\\]");
+        assertQueryFails(format("CREATE TABLE %s.%s (a_number INT) WITH (change_data_feed_enabled = true, writer_version = 5)", SCHEMA, tableName),
+                "writer_version must be an integer between \\[2, 4\\]");
+
+        assertUpdate(format("CREATE TABLE %s.%s (a_number INT) WITH (reader_version = 2)", SCHEMA, tableName));
+        assertThat((String) computeScalar("SHOW CREATE TABLE " + tableName))
+                .matches(Pattern.compile(".*(reader_version = 2,(.*)writer_version = 2).*", Pattern.DOTALL)); // default writer version
+        assertUpdate("DROP TABLE " + tableName);
+
+        assertUpdate(format("CREATE TABLE %s.%s (a_number INT) WITH (writer_version = 4)", SCHEMA, tableName));
+        assertThat((String) computeScalar("SHOW CREATE TABLE " + tableName))
+                .matches(Pattern.compile(".*(reader_version = 1,(.*)writer_version = 4).*", Pattern.DOTALL)); // default reader version
+        assertUpdate("DROP TABLE " + tableName);
+
+        assertUpdate(format("CREATE TABLE %s.%s (a_number INT) WITH (reader_version = 2, writer_version = 3)", SCHEMA, tableName));
+        assertThat((String) computeScalar("SHOW CREATE TABLE " + tableName))
+                .matches(Pattern.compile(".*(reader_version = 2,(.*)writer_version = 3).*", Pattern.DOTALL));
+        assertUpdate("DROP TABLE " + tableName);
+
+        assertUpdate(format("CREATE TABLE %s.%s (a_number INT) WITH (change_data_feed_enabled = true)", SCHEMA, tableName));
+        assertThat((String) computeScalar("SHOW CREATE TABLE " + tableName))
+                .matches(Pattern.compile(".*(change_data_feed_enabled = true,(.*)reader_version = 1,(.*)writer_version = 4).*", Pattern.DOTALL));
+        assertUpdate("DROP TABLE " + tableName);
+
+        assertUpdate(format("CREATE TABLE %s.%s (a_number INT) WITH (change_data_feed_enabled = true, writer_version = 3)", SCHEMA, tableName));
```

Comment:
a separate test with a meaningful name would allow us to avoid adding this comment.